### PR TITLE
docs(*): add osm namespace monitoring doc

### LINKED ISSUE #2904: doc: Explain how the "osm namespace add|remove" command works
Create a document explaining how the "osm namespace add|remove" command works and what workarounds exists when/if it does not work.

### PR DESCRIPTION
**Pull request by michelleN**
resolves #2904

Signed-off-by: Michelle Noorali <minooral@microsoft.com>

Please mark with X for applicable areas.

-->
**Affected area**:

- New Functionality      [ ]
- Documentation          [X ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Demo                   [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution? No

### CODE REVIEW

File: docs/content/docs/tasks_usage/namespace_monitoring.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+## Adding a Namespace to the OSM Control Plane
+
+Add a namespace for monitoring and sidecar injection to the mesh with the following command:
+
+```bash
+osm namespace add <mesh-name>
+```
+
+Explicitly disable sidecar injection while adding the namespace using `--disable-sidecar-injection flag as show [here](../sidecar_injection.md#Explicitly-Disabling-Automatic-Sidecar-Injection-on-Namespaces).
```
Comment (SanyaKochhar) [inline docs/content/docs/tasks_usage/namespace_monitoring.md:28]: The link seems to be giving a 404 for me. Not sure if it's going one dir too far back?
```suggestion
Explicitly disable sidecar injection while adding the namespace using `--disable-sidecar-injection flag as show [here](./sidecar_injection.md#Explicitly-Disabling-Automatic-Sidecar-Injection-on-Namespaces).
```

File: docs/content/docs/tasks_usage/namespace_monitoring.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+```bash
+osm namespace remove <mesh-name>
+```
+
+This command will remove the OSM specific labels and annotations on the namespace thus removing it from the mesh.
+
+## Enable Metrics for a Namespace
+
+```bash
+osm metrics enable --namespace <namespace>
```
Comment (SanyaKochhar) [inline docs/content/docs/tasks_usage/namespace_monitoring.md:43]: just curious as to why "enable" is showing up blue... made me feel like I should be clicking a link here

File: docs/content/docs/tasks_usage/namespace_monitoring.md (hunk 1 above)
Comment (michelleN) [reply docs/content/docs/tasks_usage/namespace_monitoring.md]: Thank you!

File: docs/content/docs/tasks_usage/namespace_monitoring.md (hunk 2 above)
Comment (michelleN) [reply docs/content/docs/tasks_usage/namespace_monitoring.md:43]: hm... I don't have an answer for that. Maybe @flynnduism knows?

File: docs/content/docs/tasks_usage/namespace_monitoring.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+```bash
+osm namespace list --mesh-name=<mesh-name>
+NAMESPACE         MESH   SIDECAR-INJECTION
+<namespace>       osm    enabled
+```
+
+If the namespace does not show up, check the labels on the namespace using `kubectl`:
+
+```bash
+$ kubectl get namespace <namespace> --show-labels
```
Comment (SanyaKochhar) [inline docs/content/docs/tasks_usage/namespace_monitoring.md:77]: nit for consistency: `$` in some commands but not all

Comment (michelleN) [reply docs/content/docs/tasks_usage/namespace_monitoring.md]: resolved. thanks

File: docs/content/docs/tasks_usage/namespace_monitoring.md (hunk 2 above)
Comment (flynnduism) [reply docs/content/docs/tasks_usage/namespace_monitoring.md:43]: The theme uses a code styling library called `prism.js` which colors and highlights code snippets. [This is the config](https://prismjs.com/download.html#themes=prism&languages=markup+css+clike+javascript+bash+c+csharp+cpp+go+java+markdown+python+scss+sql+toml+yaml&plugins=toolbar+copy-to-clipboard) that is included in the docs theme (the prism default, and is parsing the snippet in question as bash). 

We could try different themes and configs if this one isn't appealing, ideally under a separate issue.

Comment (michelleN) [reply docs/content/docs/tasks_usage/namespace_monitoring.md:43]: thanks @flynnduism

Comment (michelleN) [reply docs/content/docs/tasks_usage/namespace_monitoring.md:43]: @SanyaKochhar would you mind opening an issue?

**Review by nojnhuh — CHANGES_REQUESTED** [review]
Looks good overall, just a few tiny nits.

**Review by SanyaKochhar — APPROVED** [review]

File: docs/content/docs/tasks_usage/namespace_monitoring.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+all resources within that namespace. The annotation `openservicemesh.io/sidecar-injection=enabled` enables OSM to automatically inject sidecar proxy containers in all Pods created within that namespace. The metrics annotation `openservicemesh.io/metrics=enabled` allows OSM to collect metrics on resources within a Namespace.
+
+See how to use the OSM CLI to manage namespace monitoring below.
+
+## Adding a Namespace to the OSM Control Plane
+
+Add a namespace for monitoring and sidecar injection to the mesh with the following command:
+
+```bash
+osm namespace add <mesh-name>
```
[the remark this answers is not in the document]
Comment (michelleN) [reply docs/content/docs/tasks_usage/namespace_monitoring.md]: thank you. updated.

File: docs/content/docs/tasks_usage/namespace_monitoring.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+```
+
+Explicitly disable sidecar injection while adding the namespace using `--disable-sidecar-injection flag as show [here](../tasks_usage/sidecar_injection.md#Explicitly-Disabling-Automatic-Sidecar-Injection-on-Namespaces).
```
[the remark this answers is not in the document]
Comment (michelleN) [reply docs/content/docs/tasks_usage/namespace_monitoring.md]: thank you updated.

File: docs/content/docs/tasks_usage/namespace_monitoring.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+## Remove a Namespace from the OSM control plane
+
+Remove a namespace from being monitored by the mesh and disable sidecar injection with the following command:
+
+```bash
+osm namespace remove <mesh-name>
```
[the remark this answers is not in the document]
Comment (michelleN) [reply docs/content/docs/tasks_usage/namespace_monitoring.md]: thank you updated

File: docs/content/docs/tasks_usage/namespace_monitoring.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+```bash
+osm metrics enable --namespace <namespace>
+```
+
+## Ignore a Namespace
+
+There may be namespaces in a cluster that should never be part of a mesh. To explicity exclude a namespace from OSM:
+
+``` bash
+osm namespace ignore <namespace>
```
[the remark this answers is not in the document]
Comment (michelleN) [reply docs/content/docs/tasks_usage/namespace_monitoring.md:51]: This causes a bug! Documented just now here https://github.com/openservicemesh/osm/issues/2965
Suggested we update the docs to describe this behavior once it is fixed.

File: docs/content/docs/tasks_usage/namespace_monitoring.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+```bash
+osm namespace list --mesh-name=<mesh-name>
+```
+
+## Troubleshooting Guide
+
+### Policy Issues
+
+If you're not seeing changes in SMI policies being applied to a resources in a namespace, ensure the namespace is enrolled in the correct mesh:
```
[the remark this answers is not in the document]
Comment (michelleN) [reply docs/content/docs/tasks_usage/namespace_monitoring.md]: thank you updated

**Review by nojnhuh — APPROVED** [review]

**Pull request merged**